src/test/java/com/epam/tc/hw2/ex1/HomePageTest.java

```diff
@@ -0,0 +1,205 @@
+        pageOpenTest();
+
+        // 3. Perform login
+        login();
+
+        // 4. Assert Username is logged
+        loginTest();
+
+        // 5. Assert that there are 4 items on the header section are displayed,
+        //    and they have proper texts
+        headerSectionItemsTest();
+
+        // 6. Assert that there are 4 images on the Index Page,
+        //    and they are displayed
+        indexPageImagesTest();
+
+        // 7. Assert that there are 4 texts on the Index Page under icons,
+        //    and they have proper text
+        indexPageTextsTest();
+
+        // 8. Assert that there is the iframe with “Frame Button” exist
+        // 9. Switch to the iframe and check that there is “Frame Button” in the iframe
+        // 10. Switch to original window back
+        iframeTest();
+
+        // 11. Assert that there are 5 items in the Left Section are displayed,
+        //     and they have proper text
+        leftSectionItemsTest();
+    }
+
+    private void openSite() {
+        driver.navigate().to(Utils.HOME_PAGE_URL);
+    }
+
+    private void pageOpenTest() {
+        SoftAssert softAssert = new SoftAssert();
+
+        softAssert.assertEquals(driver.getCurrentUrl(), Utils.HOME_PAGE_URL);
+        softAssert.assertEquals(driver.getTitle(), Utils.HOME_PAGE_TITLE);
+        softAssert.assertAll();
+    }
+
+    private void login() {
+        WebElement loginButton = wait.until(ExpectedConditions.elementToBeClickable(
+            By.xpath("//a[@href='#']")
+        ));
+        loginButton.click();
+        WebElement loginField = wait.until(ExpectedConditions.presenceOfElementLocated(
+            By.id("name")
+        ));
+        loginField.sendKeys(Utils.LOGIN);
+        WebElement passwordField = wait.until(ExpectedConditions.presenceOfElementLocated(
+            By.id("password")
+        ));
+        passwordField.sendKeys(Utils.PASSWORD);
+        WebElement submitButton = wait.until(ExpectedConditions.elementToBeClickable(
+            By.id("login-button")
+        ));
+        submitButton.click();
+    }
+
+    private void loginTest() {
+        String actualUsername = wait.until(ExpectedConditions.presenceOfElementLocated(
+            By.id("user-name"))
+        ).getText();
+
+        SoftAssert softAssert = new SoftAssert();
+        softAssert.assertEquals(actualUsername, Utils.USERNAME);
+        softAssert.assertAll();
+    }
+
+    private void headerSectionItemsTest() {
+        SoftAssert softAssert = new SoftAssert();
+
+        List<WebElement> menuButtons = wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(
+            By.cssSelector(".nav > li")
+        ));
+
+        int num = menuButtons.size();
+        softAssert.assertEquals(num, Utils.MENU_BUTTONS_NUMBER);
+
+        for (int i = 0; i < menuButtons.size(); i++) {
+            softAssert.assertEquals(menuButtons.get(i).getText(),
+                Utils.menuButtons.get(i));
+        }
```

Comment:
Lets try to convert menuButtons to list of strings and compare like `assertThat(actualStringCollection).isEqualTo(expectedStringCollection)`